scripts/generate_shacl_tests.py

```diff
@@ -0,0 +1,366 @@
+        instance = BNode()
+
+        # For node shapes
+        if constraints["targetClass"]:
+            g.add((instance, RDF.type, constraints["targetClass"]))
+
+        # For property shapes
+        if constraints["path"]:
+            parent = BNode()
+            if constraints["datatype"]:
+                value = self._generate_valid_value(constraints["datatype"])
+                g.add((parent, constraints["path"], value))
+            elif constraints["class"]:
+                obj = BNode()
+                g.add((obj, RDF.type, constraints["class"]))
+                g.add((parent, constraints["path"], obj))
+            else:
+                g.add((parent, constraints["path"], Literal("Valid value")))
+            return g, parent
+
+        # Handle nested properties
+        for prop in constraints["properties"]:
+            path = prop["path"]
+            min_count = int(prop["minCount"].value) if prop["minCount"] else 1
+
+            for _ in range(min_count):
+                if prop["datatype"]:
+                    value = self._generate_valid_value(prop["datatype"])
+                    g.add((instance, path, value))
+                elif prop["class"]:
+                    obj = BNode()
+                    g.add((obj, RDF.type, prop["class"]))
+                    g.add((instance, path, obj))
+                else:
+                    g.add((instance, path, Literal(f"Valid value for {path}")))
+
+        return g, instance
+
+    def _generate_valid_value(self, datatype: URIRef) -> Literal:
+        """Generate a valid value for the given datatype."""
+        if datatype == XSD.string:
+            return Literal("Valid string", datatype=XSD.string)
+        elif datatype == XSD.dateTime:
+            return Literal("2024-03-20T12:00:00", datatype=XSD.dateTime)
+        elif datatype == XSD.date:
+            return Literal("2024-03-20", datatype=XSD.date)
+        elif datatype == XSD.integer:
+            return Literal("42", datatype=XSD.integer)
+        elif datatype == XSD.decimal:
+            return Literal("42.0", datatype=XSD.decimal)
+        elif datatype == XSD.boolean:
+            return Literal("true", datatype=XSD.boolean)
+        return Literal("Default value", datatype=datatype)
+
+    def _get_shape_name(self, shape: URIRef) -> str:
+        """Get the standardized name for a shape from the SHACL file."""
+        # First try to get the shape name from the URI
+        uri = str(shape)
+        if "#" in uri:
+            # Handle URIs like http://data.europa.eu/a4g/data-shape#dcat-Distribution-dcat-accessURL
+            return uri.split("#")[-1]
+
+        # Fallback to label if URI doesn't follow the pattern
+        label = self.shapes_graph.value(shape, RDFS.label)
+        if label:
+            return str(label)
+
+        # Last resort: use the last part of the URI
+        return uri.split("/")[-1]
+
+    def generate_invalid_instances(
+        self, shape: URIRef, constraints: Dict
```

Comment:
I would be batter that constraints being a model instead of dict. But it's ok while experimental.